core-management/core-management-client/src/main/java/org/wildfly/extension/core/management/client/ProcessStateListener.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.extension.core.management.client;
+
+import java.util.Map;
+
+/**
+ * @author <a href="http://jmesnil.net/">Jeff Mesnil</a> (c) 2016 Red Hat inc.
+ */
+public interface ProcessStateListener {
+
+    /**
+     * Initialize the listener.
+     * This should not throw an exception.
+     * @param properties the properties to initialize the listener.
+     */
+    default void init(Map<String, String> properties) {}
```

Comment:
**If** we do this, we should make sure that we wrap this in a class (ProcessStateListenerInitEvent or something like that) so that we can extend it with more information in future